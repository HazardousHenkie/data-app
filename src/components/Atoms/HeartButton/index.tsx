import React from 'react'

import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder'
import IconButton from '@material-ui/core/IconButton'

interface HeartButtonProps {
    label: string
    heartOnClick: () => void
}

const HeartButton: React.FC<HeartButtonProps> = ({ label, heartOnClick }) => {
    return (
        <IconButton onClick={heartOnClick} aria-label={label}>
            <FavoriteBorderIcon />
        </IconButton>
    )
}

export default HeartButton
