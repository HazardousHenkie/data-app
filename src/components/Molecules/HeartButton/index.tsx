import React from 'react'

import IconButton from '@material-ui/core/IconButton'

import InlineLoader from 'components/Atoms/InlineLoader'

import HeartButtonWrapper, {
    FavoriteIconStyled,
    FavoriteBorderIconStyled
} from './styledComponents'

interface HeartButtonProps {
    loading: boolean
    active: boolean
    label: string
    heartOnClick: () => void
}

const HeartButton: React.FC<HeartButtonProps> = ({
    loading,
    active,
    label,
    heartOnClick
}) => {
    return (
        <HeartButtonWrapper>
            {loading && <InlineLoader />}

            <IconButton
                onClick={heartOnClick}
                aria-label={label}
                data-testid="iconButton"
            >
                {active ? (
                    <FavoriteIconStyled data-testid="favoriteIcon" />
                ) : (
                    <FavoriteBorderIconStyled data-testid="favoriteIconBorder" />
                )}
            </IconButton>
        </HeartButtonWrapper>
    )
}

export default HeartButton
