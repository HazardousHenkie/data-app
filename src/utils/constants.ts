enum ENDPOINTS {
    LOGOUT = '/.netlify/functions/googleLogout',
    lOGIN = '/.netlify/functions/googleLogin',
    REFRESH_TOKEN = '/.netlify/functions/refreshToken',
    COUNTRIES = '/.netlify/functions/countries',
    FAVORITE_COUNTRIES = '/.netlify/functions/getFavoritedCountries',
    SAVE_COUNTRY = '/.netlify/functions/saveCountry/',
    DELETE_COUNTRY = '/.netlify/functions/deleteCountry/'
}

export default ENDPOINTS
