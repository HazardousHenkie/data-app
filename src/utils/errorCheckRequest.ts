import { put } from 'redux-saga/effects'
import { getRefreshTokenRequest } from 'globals/authentication/refreshToken/actions'
import { logoutRequest } from 'globals/authentication/logout/actions'

import { setError } from 'globals/globalErrors/actions'
import { ResponseError } from './request'

// test this one?
function* requestErrorCheck(error: ResponseError) {
    try {
        if (localStorage.getItem('userId')) {
            yield put(
                getRefreshTokenRequest(localStorage.getItem('userId') as string)
            )
        } else {
            throw Error('No userId in local storage.')
        }
    } catch {
        yield put(logoutRequest())
        yield put(setError(error))
    }
}
export default requestErrorCheck
