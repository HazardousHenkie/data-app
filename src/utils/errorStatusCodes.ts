enum ERROR_STATUS_CODES {
    BAD_REQUEST = 400,
    UNAUTHORIZED = 401,
    SERVER_ERROR = 500,
    SERVICE_UNAVAILABLE = 503,
    NOT_FOUND = 404
}

export default ERROR_STATUS_CODES
