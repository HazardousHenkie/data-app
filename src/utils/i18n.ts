import i18n from 'i18next'
import Backend from 'i18next-xhr-backend'
import LanguageDetector from 'i18next-browser-languagedetector'
import { initReactI18next } from 'react-i18next'

export const appLocales = ['en', 'ja', 'nl']

i18n.use(Backend)
    .use(Backend)
    .use(LanguageDetector)
    .use(initReactI18next)
    .init({
        debug: false,
        lng: 'en',
        fallbackLng: 'en',
        whitelist: appLocales,
        interpolation: {
            escapeValue: false
        },
        react: {
            wait: true
        }
    })

export default i18n
