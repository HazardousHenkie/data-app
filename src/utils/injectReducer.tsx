import * as React from 'react'
import hoistNonReactStatics from 'hoist-non-react-statics'
import { useStore, ReactReduxContext } from 'react-redux'

import { getInjectors } from './reducerInjectors'
import { InjectReducerParams, InjectedStore } from 'types'

export default function hocWithReducer<P>({
  key,
  reducer
}: InjectReducerParams) {
  function wrap(
    WrappedComponent: React.ComponentType<P>
  ): React.ComponentType<P> {
    class ReducerInjector extends React.Component<P> {
      public static contextType = ReactReduxContext
      public static WrappedComponent = WrappedComponent
      public static displayName = `withReducer(${WrappedComponent.displayName ||
        WrappedComponent.name ||
        'Component'})`

      constructor(props: any, context: any) {
        super(props, context)

        getInjectors(context.store).injectReducer(key, reducer)
      }

      public render() {
        return <WrappedComponent {...this.props} />
      }
    }

    return hoistNonReactStatics(ReducerInjector, WrappedComponent) as any
  }
  return wrap
}

const useInjectReducer = ({ key, reducer }: InjectReducerParams) => {
  const store = useStore() as InjectedStore
  React.useEffect(() => {
    getInjectors(store).injectReducer(key, reducer)
  }, [key, reducer, store])
}

export { useInjectReducer }
