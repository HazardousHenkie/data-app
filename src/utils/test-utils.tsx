import React from 'react'
import { render } from '@testing-library/react'
import configureStore from 'configureStore'

import history from 'utils/history'
import { Provider } from 'react-redux'

import variables from 'styles/variables'
import { ThemeProvider } from 'styled-components'
import { ThemeProvider as MuiThemeProvider } from '@material-ui/styles'
import lightTheme from 'styles/themeStyles'

const customRender = (
    ui: React.ReactElement,
    {
        initialState = {},
        store = configureStore(initialState, history),
        ...renderOptions
    } = {}
) => {
    const Wrapper: React.FC = ({ children }) => {
        return (
            <Provider store={store}>
                <ThemeProvider theme={variables}>
                    <MuiThemeProvider theme={lightTheme}>
                        <ThemeProvider theme={lightTheme}>
                            {children}
                        </ThemeProvider>
                    </MuiThemeProvider>
                </ThemeProvider>
            </Provider>
        )
    }
    return render(ui, {
        wrapper: Wrapper,
        ...renderOptions
    })
}

export * from '@testing-library/react'

export { customRender as render }
