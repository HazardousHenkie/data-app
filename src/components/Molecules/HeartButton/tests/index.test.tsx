import React from 'react'

import { render } from 'utils/test-utils'

import HeartButton from '../index'

const heartComponent = (
    <HeartButton loading={false} active label="label" heartOnClick={() => {}} />
)

describe('<HeartButton />', () => {
    it('should render like snapshot', () => {
        const component = render(heartComponent)

        expect(component).toMatchSnapshot()
    })

    it('should render the wrapper component with a div', () => {
        const { container } = render(heartComponent)

        expect(container.tagName).toEqual('DIV')
    })

    it('it should render a button because there is an onclick', () => {
        const { getByTestId } = render(heartComponent)

        const button = getByTestId('iconButton')

        expect(button.tagName).toEqual('BUTTON')
    })

    it('it should render the IconButton and aria label', () => {
        const { getByLabelText } = render(heartComponent)

        const button = getByLabelText('label')

        expect(button).toBeInTheDocument()
    })

    it('it should render the FavoriteIconStyled component if active is true', () => {
        const { getByTestId } = render(heartComponent)

        const icon = getByTestId('favoriteIcon')
        expect(icon).toBeInTheDocument()
    })

    it('it should render the FavoriteBorderIconStyled component if active is false', () => {
        const { getByTestId } = render(
            <HeartButton
                loading
                active={false}
                label="label"
                heartOnClick={() => {}}
            />
        )

        const icon = getByTestId('favoriteIconBorder')
        expect(icon).toBeInTheDocument()
    })

    test('Show loader when loading is true', () => {
        const { getByTestId } = render(
            <HeartButton loading active label="label" heartOnClick={() => {}} />
        )

        const loadingComponent = getByTestId('inlineLoader')

        expect(loadingComponent).toBeInTheDocument()
    })

    test("Don't show loader when loading is false", () => {
        const { queryByTestId } = render(heartComponent)

        const loadingComponent = queryByTestId('inlineLoader')

        expect(loadingComponent).toBeFalsy()
    })
})
