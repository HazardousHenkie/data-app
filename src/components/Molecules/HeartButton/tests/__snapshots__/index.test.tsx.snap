// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeartButton /> should render like snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxirZ kNDIFW"
      >
        <button
          aria-label="label"
          class="MuiButtonBase-root MuiIconButton-root"
          data-testid="iconButton"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root sc-AxiKw dxxXig"
              data-testid="favoriteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxirZ kNDIFW"
    >
      <button
        aria-label="label"
        class="MuiButtonBase-root MuiIconButton-root"
        data-testid="iconButton"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root sc-AxiKw dxxXig"
            data-testid="favoriteIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 21.35l-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
